samcli/lib/providers/cfn_api_provider.py

```diff
@@ -75,6 +95,208 @@ def extract_resources(self, stacks: List[Stack], collector: ApiCollector, cwd: O
+        collector: ApiCollector
+            ApiCollector to save Authorizers into
+        """
+        properties = resource.get("Properties", {})
+        authorizer_type = properties.get(CfnApiProvider._AUTHORIZER_TYPE, "").lower()
+        rest_api_id = properties.get(CfnApiProvider._AUTHORIZER_REST_API)
+        name = properties.get(CfnApiProvider._AUTHORIZER_NAME)
+        authorizer_uri = properties.get(CfnApiProvider._AUTHORIZER_AUTHORIZER_URI)
+
+        if not authorizer_type:
+            raise InvalidSamTemplateException(
+                f"Authorizer '{logical_id}' is missing the '{CfnApiProvider._AUTHORIZER_TYPE}' "
+                "property, an Authorizer type must be defined."
+            )
+
+        if not rest_api_id:
+            raise InvalidSamTemplateException(
+                f"Authorizer '{logical_id}' is missing the '{CfnApiProvider._AUTHORIZER_REST_API}' "
+                "property, this must be defined."
+            )
+
+        if not name:
+            raise InvalidSamTemplateException(
+                f"Authorizer '{logical_id}' is missing the '{CfnApiProvider._AUTHORIZER_NAME}' "
+                "property, the Name must be defined."
+            )
+
+        if authorizer_type not in LambdaAuthorizer.VALID_TYPES:
+            LOG.warning(
+                "Authorizer '%s' with type '%s' is currently not supported. "
+                "Only Lambda Authorizers of type TOKEN and REQUEST are supported.",
+                logical_id,
+                authorizer_type,
+            )
+            return
+
+        if not authorizer_uri:
+            raise InvalidSamTemplateException(
+                f"Authorizer '{logical_id}' is missing the '{CfnApiProvider._AUTHORIZER_AUTHORIZER_URI}' "
+                "property, a valid Lambda ARN must be provided."
+            )
+
+        function_name = LambdaUri.get_function_name(authorizer_uri)
+        if not function_name:
+            LOG.warning(
+                "Was not able to resolve Lambda function ARN for Authorizer '%s'. "
+                "Double check the ARN format, or use more simple intrinsics.",
+                logical_id,
+            )
+            return
+
+        identity_source_template = properties.get(CfnApiProvider._AUTHORIZER_IDENTITY_SOURCE, None)
+
+        if identity_source_template is None and authorizer_type == LambdaAuthorizer.TOKEN:
+            raise InvalidSamTemplateException(
+                f"Lambda Authorizer '{logical_id}' of type TOKEN, must have "
+                f"'{CfnApiProvider._AUTHORIZER_IDENTITY_SOURCE}' of type string defined."
+            )
+
+        if identity_source_template is not None and not isinstance(identity_source_template, str):
+            raise InvalidSamTemplateException(
+                f"Lambda Authorizer '{logical_id}' contains an invalid '{CfnApiProvider._AUTHORIZER_IDENTITY_SOURCE}', "
+                "it must be a comma-separated string."
+            )
```

Comment:
For authorizers of type `REQUEST`, identity sources are optional in API Gateway V1 depending on if caching is enabled or not (if caching is enabled, must provide identity sources). I've made the choice to not validate this.

Reply:
It is great to read these feedbacks, but I wonder if we should write some of them in the code so that we can understand the decision later? 🤔